README.rst

```diff
@@ -1,7 +1,44 @@
 Byterun
 -------
 
-This is a pure-Python implementation of a Python bytecode execution virtual
-machine.  I started it to get a better understanding of bytecodes so I could
-fix branch coverage bugs in coverage.py.
+
+
+Byterun is a python interpreter written in just 1451 lines of Python.
+Byterun's goal is to clearly explain Python's design. 
+If you want to better understand Python, or you are not quite sure how something in cPYthon works, then Byterun is  a great place to start. 
+Byterun has proven very useful to a number of different projects, `including one by Google <https://github.com/nedbat/byterun/pull/12>`_.  
+
+By interpreter we mean a code execution virtual 
+machine.  The Python compiler converts python source code into Python bytecode.  The code execution virtual machine then executes that bytecode.   
+
+There are a number of python virtual machines.  Most Python developers use cPython.  cPython includes a virtual machine written in C.  Yes it runs Python very fast, but it is a large code base, and difficult to understand.  Much better to start by studying Byterun.  Then you can move onto cPython.   
+
+If you are interested in Byterun,  the first step is to read the 
+excellent introductory article on Byterun written by Alison 
+Kaptur.  `A Python Interpreter written in Python <http://www.aosabook.org/en/500L/a-python-interpreter-written-in-python.html>`_
+
+If you are not sure what a stack machine is, you might even start  
```

Comment:
We may want to write a small explanation of a stack machine in our own docs. I find that with CS concepts on wikipedia it can be hard to tease apart abstract concepts and things that are particular to some runtime/language that the author knows.